### DISCUSSION TITLE
Kitty is 60fps confirmed

### DISCUSSION THREAD

**Discussion by ctrlcctrlv** [kovidgoyal/kitty#3673] [Show and tell]
`kitty` once again proves it's the best terminal emulator for Unix-likes~

https://twitter.com/fr_brennan/status/1398875075873492993

![image](https://user-images.githubusercontent.com/838783/120093366-bf09c900-c0e7-11eb-8bef-a6b9fc15d88c.png)
![image](https://user-images.githubusercontent.com/838783/120093362-b74a2480-c0e7-11eb-81e0-0b711e453824.png)

**Comment by kovidgoyal** [comment]
Cool, if someday mpv ever gets support for the kitty graphics protocol you should be able to enjoy pixel perfect 60fps video in your terminal, even over ssh (assuming your network link is fast enough).

**Reply by ctrlcctrlv** [reply]
@kovidgoyal It has sixel support…remember how we talked about this before? I'm looking into dusting off my patch right now.

I agree Kitty graphics would be better, but want to see sixels in Kitty! :-)

**Reply by ctrlcctrlv** [reply]
Here's how the sixels look like in xterm.

![image](https://user-images.githubusercontent.com/838783/120093702-e6fa2c00-c0e9-11eb-858b-dbf728965e23.png)

xterm has a 1000x1000 pixel limit on its sixel graphics, which we can easily beat.

I'm inspired to work on this…let's see how far I get tonight. I've got all night.

**Reply by kovidgoyal** [reply]
Enjoy :) Although given the history of the terminal space, am pretty sure if you add sixels support to kitty, the entire ecosystem will just settle on sixels, and we will be left with pathetic graphics in terminals forever. 

Still, as I said, I wont refuse a patch, if you can actually get sixels (which have some crazy semantics, just ask @dankamongmen) to work on top of kitty's graphics layer, more power to you!

**Reply by ctrlcctrlv** [reply]
To help prevent that @kovidgoyal I'm going to depend on `libsixel`, which is not even packaged on Arch (it's an AUR dependency). Meaning, we'll be disabling this feature by default. Meaning, there's still going to be pain involved in activating sixels (similar to the fact that `xterm` requires a flag to even listen for them). Meanwhile, implementors can guarantee graphics will display if they use Kitty graphics.

**Reply by dankamongmen** [reply]
the kitty protocol isn't perfect, but it's wildly superior to the sixel "protocol" (lack of standardization across sixel implementations makes me hesitant to call it such). if you intend to add sixel to kitty, how do you intend to have the two interact? would a sixel blow away kitty graphics data? would it depend on the z-level of the latter? what about a partially-transparent kitty graphic atop a sixel? does it blow it away at cell-granularity, as glyphs do? there are a lot of subtle issues here.

i offered a patch to mpv to use notcurses to get portable support for terminal graphics (kitty+sixel), but they were rather hostile to the idea: https://github.com/mpv-player/mpv/issues/8344

know that libsixel is, AFAICT, no longer maintained and has active bugs: https://github.com/saitoha/libsixel/issues/149

i'm not trying to dump on your idea, but...well, i kinda am. why not just support the kitty protocol in whatever media player you're hacking on? it's quite possible to determine support for sixel or kitty at runtime -- check Notcurses if you have doubts.

**Comment by ctrlcctrlv** [comment]
And now in Sixel:

https://twitter.com/fr_brennan/status/1399974490881265664
